cpp/src/io/json/nested_json_gpu.cu

```diff
@@ -0,0 +1,804 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "nested_json.hpp"
+
+#include <io/fst/logical_stack.cuh>
+#include <io/fst/lookup_tables.cuh>
+#include <io/utilities/hostdevice_vector.hpp>
+
+#include <cudf/types.hpp>
+#include <cudf/utilities/span.hpp>
+
+#include <rmm/exec_policy.hpp>
+
+namespace cudf::io::json {
+
+//------------------------------------------------------------------------------
+// JSON-TO-STACK-OP DFA
+//------------------------------------------------------------------------------
```

Comment:
Header lines are a little aggressive. Also it would be nice to have a definition of DFA somewhere in the code, since that abbreviation is used repeatedly.
```suggestion
// JSON to stack operator DFA (Deterministic Finite Automata)```